### PR TITLE
add method GetFile and refactor code

### PR DESCRIPTION
**Pull request by robatipoor**
some time we need work with file in memory not file system

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -214,3 +216,16 @@ func thumbnailToFilemap(thumb *Photo) map[string]File {
 	}
 	return nil
 }
+
+func get(url string) ([]byte, error) {
```
Comment (stek29) [inline util.go]: I'd suggest a more descriptive name

File: bot.go [hunk 2]
```diff
@@ -877,36 +876,49 @@ func (b *Bot) FileByID(fileID string) (File, error) {
...
+	url := fmt.Sprintf("%s/file/bot%s/%s",
+		b.URL, b.Token, f.FilePath)
 
-	return nil
+	out, err := get(url)
+	if err != nil {
+		return nil, wrapSystem(err)
+	}
+
+	*file = f
```
Comment (stek29) [inline bot.go:917]: What's the purpose of this assignment?

File: bot.go [hunk 3]
```diff
@@ -877,36 +876,49 @@ func (b *Bot) FileByID(fileID string) (File, error) {
...
 
-	_, err = io.Copy(out, resp.Body)
+	f.FileLocal = localFileName
+	*file = f
+
+	return nil
+}
+
+// GetFile from Telegram servers
+func (b *Bot) GetFile(file *File) ([]byte, error) {
```
Comment (stek29) [inline bot.go]: Why won't you reuse GetFile in download?

File: bot.go [hunk 4]
```diff
@@ -877,36 +876,49 @@ func (b *Bot) FileByID(fileID string) (File, error) {
...
 
-	out, err := os.Create(localFilename)
+	out, err := get(url)
 	if err != nil {
 		return wrapSystem(err)
 	}
-	defer out.Close()
 
-	resp, err := http.Get(url)
+	err = ioutil.WriteFile(localFileName, out, 0644)
```
Comment (stek29) [inline bot.go]: Old code was doing io.Copy from resp.Body, and didn't read everything into memory.
I'd say it's a regression.

**Review by stek29 — CHANGES_REQUESTED** [review]

**Comment by robatipoor** [comment]
please review pull request again

**Review by stek29 — APPROVED** [review]
lgtm

**Comment by tucnak** [comment]
Thank you! Sorry for a late merge.

**Pull request merged**